pkg/util/httputil/decoratedclient_test.go

```diff
@@ -0,0 +1,24 @@
+package httputil
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+)
+
+func Test_DecorateClient(t *testing.T) {
+	httpClient := &http.Client{Transport: http.DefaultTransport}
+	withDigest := Digest("publiApi", "privateApi")
+	withLogging := LoggingTransport(zap.S())
+
+	decorated, err := DecorateClient(&http.Client{Transport: http.DefaultTransport}, withDigest, withLogging)
+	assert.NoError(t, err)
+	assert.Equal(t, httpClient.Timeout, decorated.Timeout)
+	assert.Equal(t, httpClient.Jar, decorated.Jar)
+	assert.NotNil(t, decorated.Transport)
```

Comment:
[nit] when asserting multiple times `testify` recommends to use the following pattern 

```suggestion
	a := assert.New(t) // using "a" as example but the name is not important 
	a.NoError(err)
	a.Equal(httpClient.Timeout, decorated.Timeout)
	a.Equal(httpClient.Jar, decorated.Jar)
	a.NotNil(decorated.Transport)
```